core/dbt/main.py

```diff
@@ -713,10 +651,11 @@ def parse_args(args):
     _build_deps_subparser(subs, base_subparser)
     _build_archive_subparser(subs, base_subparser)
 
+    rpc_sub = _build_rpc_subparser(subs, base_subparser)
     run_sub = _build_run_subparser(subs, base_subparser)
     compile_sub = _build_compile_subparser(subs, base_subparser)
     generate_sub = _build_docs_generate_subparser(docs_subs, base_subparser)
-    _add_common_arguments(run_sub, compile_sub, generate_sub)
+    _add_common_arguments(run_sub, compile_sub, generate_sub, rpc_sub)
```

Comment:
are the common args like `--models`, `--exclude`, `--non-destructive`, and `--full-refresh` applicable for the rpc task? I imagine that `--threads` and `--no-version-check` might be

Reply:
yup, that's what this does!

Reply:
ok! I think we probably don't want `--models`, `--exclude`, `--non-destructive`, and `--full-refresh` to be acceptable flags to this task. I don't think they're sensible in the context of a client <> server paradigm. Can we remove these?

I could buy the use case for `--threads` and `--no-version-check` though.

Reply:
We can, and since `--models`, `--exclude` are both ignored, removing them won't change anything.

It's worth pointing out that although I'm not sure of the ultimate consequences, `--full-refresh` and `--non-destructive` do have impacts on the execution of code in the dbt rpc server model! It flips a global flag value stored in `dbt.flags` that's then accessible to macros and such, and some adapter methods also inspect them. I think the solution here is probably that `flags` should be an object generated at runtime based on config/args and passed around as needed, rather than a module with attributes that are set during argument parsing. And then we should accept them in RPC calls, probably.

I think `--no-version-check` should probably be available everywhere that loads project files (which the server does), and we do actually use the profile's `threads` in the server, so it makes sense to keep the cli override around.